django/forms/boundfield.py

```diff
@@ -96,9 +96,18 @@ def as_widget(self, widget=None, attrs=None, only_initial=False):
             attrs.setdefault(
                 "id", self.html_initial_id if only_initial else self.auto_id
             )
+
+        if only_initial and self.html_initial_name in self.form.data:
+            value = self.form._widget_data_value(
+                self.field.hidden_widget(),
+                self.html_initial_name,
+            )
+        else:
+            value = self.value()
```

Comment:
Do you think a comment on this block might be handy? 

```
# Ensure hidden initial value is not replaced with submitted value. 
...
```

... or such. 

🤔

Reply:
Yep sounds good… if you have to ask then it's a sign it's unclear 👍

Reply:
I've taken the liberty of framing the comment in the positive:

```
# Propagate the hidden initial value.
```

How does the that sound?